docs/src/cli/deploy-a-program.md

```diff
@@ -300,3 +300,57 @@ solana program deploy --program-id <PROGRAM_ADDRESS> --buffer <BUFFER_ADDRESS>
 Note, the buffer's authority must match the program's upgrade authority.
 
 Buffers also support `show` and `dump` just like programs do.
+
+### Upgrading program using offline signer as authority
+
+Some security models require separating the signing process from the transaction broadcast, such that the signing keys can be completely disconnected from any network, also known as [offline signing](../offline-signing.md).
+
+This section describes how a program developer can use offline signing to upgrade their program, unlike the [previous section](deploy-a-program.md#redeploy-a-program), which assumes the machine is connected to the internet, aka online signing.
+
+Note that only the `upgrade` command can be performed in offline mode. The initial program deployment **must** be performed from an online machine, and only subsequent program upgrades can leverage offline signing.
+
+Assuming your program has been deployed and its upgrade authority has been changed to an
+offline signer, 
+a typical setup would consist of 3 different signers (pairs of keys):
```

Comment:
micro-nit after the change:
```suggestion
a typical setup would consist of 2 different signers:
```

Reply:
Applied in https://github.com/solana-labs/solana/pull/33860/commits/38c85c292aace6bcfcc3ce9614024ef527744c61